commons/src/main/java/org/infinispan/commons/util/concurrent/LifecycleRejectedExecutionHandler.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.commons.util.concurrent;
+
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import org.infinispan.commons.IllegalLifecycleStateException;
+
+/**
+ * A handler for rejected tasks that always throws a {@link IllegalLifecycleStateException}.
+ * @author wburns
+ * @since 11.0
+ */
+public class LifecycleRejectedExecutionHandler implements RejectedExecutionHandler {
+   private LifecycleRejectedExecutionHandler() { }
+
+   private static final LifecycleRejectedExecutionHandler INSTANCE = new LifecycleRejectedExecutionHandler();
+
+   public static LifecycleRejectedExecutionHandler getInstance() {
+      return INSTANCE;
+   }
+
+   @Override
+   public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+      throw new IllegalLifecycleStateException();
```

Comment:
Shouldn't we throw an `IllegalLifecycleStateException` only if the executor has been shut down?

Reply:
I took a peek at `ScheduledThreadPoolExecutor` and it uses an effectively unbounded queue (Integer.MAX_VALUE, which isn't configurable) and has a rejection handler that is `AbortPolicy` by default, which always throws the exception. So I figured this was good.

Reply:
Ok, but please add a comment to document the decision @wburns

Reply:
It didn't seem especially pertinent to this class myself, but okay.

Reply:
To me it seems better to document it in the class then at the call site. Somehow comments between method arguments always rub me the wrong way, even when they're not on the same line as the parameters :)